xfel/merging/application/filter/experiment_filter.py

```diff
@@ -69,18 +60,89 @@ def check_cluster(self, experiment):
     return m_distance < self.params.filter.unit_cell.cluster.covariance.mahalanobis
 
   def run(self, experiments, reflections):
-    if 'unit_cell' not in self.params.filter.algorithm: # so far only "unit_cell" algorithm is supported
+    filter_by_unit_cell = 'unit_cell' in self.params.filter.algorithm[0]
+    filter_by_n_obs = 'n_obs' in self.params.filter.algorithm[0]
+    filter_by_resolution = 'resolution' in self.params.filter.algorithm[0]
+    # only "unit_cell" "n_obs" and "resolution" algorithms are supported
+    if (not filter_by_unit_cell) and (not filter_by_n_obs) and (not filter_by_resolution):
```

Comment:
These validation steps should likely be in the validate function.  validate runs before the program starts execution, which will save time if the parameters are invalid.